staging/src/k8s.io/client-go/tools/cache/the_real_fifo.go

```diff
@@ -175,22 +187,32 @@ func (f *RealFIFO) addToItems_locked(deltaActionType DeltaType, skipTransform bo
 // Add inserts an item, and puts it in the queue. The item is only enqueued
 // if it doesn't already exist in the set.
 func (f *RealFIFO) Add(obj interface{}) error {
+	var size int
+	defer func() {
+		f.metrics.numberOfQueuedItem.Set(float64(size))
```

Comment:
general comment: doing the metric sets outside the lock means the number can go up and then down, right?

Reply:
We copy out the value from under lock, so there could be a race between `Pop` and `Add`. However I don't think we care much as the value is meant to be an approximate of size allowing to monitor trends.